features/breed_list/src/test/java/com/example/catbreeds/breed_list/unitTests/BreedListViewModelTest.kt

```diff
@@ -0,0 +1,216 @@
+        every { breedRepository.getFavoriteBreeds() } returns flowOf(emptyList())
+        coEvery { breedRepository.refreshBreeds() } returns Unit
+
+        // WHEN
+        val vm = vmUnderTest
+        advanceUntilIdle()
+
+        // THEN
+        assertEquals(testBreeds, vm.breeds.value)
+        assertEquals(testBreeds, vm.filteredBreeds.value)
+    }
+
+    @Test
+    fun `WHEN initialized SHOULD call refreshBreeds`() = runTest {
+        // GIVEN
+        every { breedRepository.getBreeds() } returns flowOf(emptyList())
+        every { breedRepository.getFavoriteBreeds() } returns flowOf(emptyList())
+        coEvery { breedRepository.refreshBreeds() } returns Unit
+
+        // WHEN
+        val vm = vmUnderTest // Triggers the lazy initialization of vmUnderTest
+        advanceUntilIdle()
+
+        // THEN
+        coVerify { breedRepository.refreshBreeds() }
+    }
+
+    @Test
+    fun `WHEN search query is updated SHOULD filter list correctly by name origin or temperament`() = runTest {
+        // GIVEN
+        val (breed1, breed2, breed3, breed4) = testBreeds
+        every { breedRepository.getBreeds() } returns flowOf(testBreeds)
+        every { breedRepository.getFavoriteBreeds() } returns flowOf(emptyList())
+        coEvery { breedRepository.refreshBreeds() } returns Unit
+        val vm = vmUnderTest
+        advanceUntilIdle()
+
+        // WHEN (search by Name segment)
+        vm.updateSearchQuery("sibe")
+        advanceUntilIdle()
+        // THEN (name match)
+        assertEquals(listOf(breed1), vm.filteredBreeds.value)
+
+        // WHEN (search by origin)
+        vm.updateSearchQuery("iran")
+        advanceUntilIdle()
+        // THEN (origin match)
+        assertEquals(listOf(breed2), vm.filteredBreeds.value)
+
+        // WHEN (search by temperament (Upper case))
+        vm.updateSearchQuery("ENERGETIC")
+        advanceUntilIdle()
+        // THEN (temperament match)
+        assertEquals(listOf(breed3), vm.filteredBreeds.value)
+
+        // WHEN (search by temperament (multiple matches))
+        vm.updateSearchQuery("Playful")
+        advanceUntilIdle()
+        // THEN (multiple matches)
+        assertEquals(2, vm.filteredBreeds.value.size)
+        assertTrue(vm.filteredBreeds.value.containsAll(listOf(breed1, breed4)))
+
+        // WHEN (no matches)
+        vm.updateSearchQuery("Zzz")
+        advanceUntilIdle()
+        // THEN (empty list)
+        assertEquals(emptyList<Breed>(), vm.filteredBreeds.value)
+    }
+
+    @Test
+    fun `WHEN toggleFavorite is called on an unfavorite breed SHOULD call addBreedToFavorites repository method`() = runTest {
+        // GIVEN
+        val breedId = testBreeds.first().id
+        every { breedRepository.getBreeds() } returns flowOf(testBreeds)
+        every { breedRepository.getFavoriteBreeds() } returns flowOf(emptyList())
+        coEvery { breedRepository.refreshBreeds() } returns Unit
```

Comment:
You can use the coJustRun on every case that you return Unit

```suggestion
        coJustRun { breedRepository.refreshBreeds() } returns Unit
```

Reply:
Made some changes on commit a2840ed